.github/workflows/docker.yml

```diff
@@ -0,0 +1,96 @@
+name: Red Hat
+on:
+  pull_request:
+  push:
+    branches: [ main ]
+jobs:
+  centos_stream:
+    name:  ${{ matrix.dist }}-${{ matrix.compiler}}-${{ matrix.build_type }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```

Comment:
This annoyingly generates a lot of combos but I guess does not hurt. How do we know when to update this matrix?

Reply:
You caught me. The purpose of this is to generate a bunch of annoying combos. :sweat_smile: 
I'm not opposed to reducing the matrix to just `Release` mode with `gcc`

Reply:
Since it is HLA ticket, I'll defer to @mdlpstsci but she might not respond until next week or two.